easy_profiler_core/event_trace_win.cpp

```diff
@@ -214,15 +258,9 @@ namespace profiler {
                 pid = GetProcessIdOfThread(hThread);
                 auto pinfo = &PROCESS_INFO_TABLE[pid];
 
-                if (pinfo->valid == 0)
```

Comment:
`valid` was used instead of `name.empty()` because sometimes `OpenProcess()` or `GetModuleBaseName()` returns 0 and we have to repeat these requests later. I faced this on Windows 7. So it is possible to get `ProcessInfo` with `valid == 0` from `map`.
Your previous changes seemed more acceptable for me...

Reply:
Oh I see. My mistake. I fixed that. However, if you still think it's too much much taken out of the function, I'll make the change more minimal. I just like minimizing function length.

Reply:
@uykusuz thanks for reply!

I have more questions :smile: 

Can't we just use `std::wstring` for `ProcessInfo::name` and leave other `processTraceEvent()` code untouched (except those `std::wstring_convert` part) ?

I don't like `std::stringstream` because of lots of memory allocations (resize buffer inside `stringstream`, return copy of `string` when calling `.str()`). 
This
```cpp
static char numbuf[128] = {};
sprintf(numbuf, "%u", pid);
pinfo->name = numbuf;
```
required only one allocation on `pinfo->name = numbuf;`.

Also when setting process name
```cpp
pinfo->name.reserve(pinfo->name.size() + 2 + len);
pinfo->name.append(" ", 1);
pinfo->name.append(buf, len);
```
there was only on allocation - `reserve(pinfo->name.size() + 2 + len);`.

Now in `set_name()` we got 2 allocations: the first one is `name += " ";` and the second one is `name.append(process_name);`.
I don't know about `std::wstring_convert<ConvertType, wchar_t>::.to_bytes(process_name)` - can we avoid additional memory allocation here or not?
Can't we use `auto len = GetModuleBaseName(hProc, 0, buf, MAX_PATH);` value to reserve necessary amount of memory as it was done before? (maybe `len * 2` for unicode case, I don't know)

So taking into account all the above, I'm returning to the first question: "Can't we just use `std::wstring` for `ProcessInfo::name` and leave other old code untouched?" :)

I'm just aiming to maximum performance.
The profiler should impact profiled application execution as least as possible.